src/yunohost/service.py

```diff
@@ -414,12 +414,20 @@ def service_regen_conf(names=[], with_diff=False, force=False, dry_run=False,
                        list_pending=False):
 
     services = _get_services()
+
+    if isinstance(names, str):
+        names = [names]
+
     for name in names:
-        if name not in services:
-            raise YunohostError('service_unknown', service=service)
+        if name not in services.keys():
```

Comment:
Misc: the `.keys()` is not needed here normally (if it's a dict "a in some_dict" == "a in some_dict.keys()"